stdlib/http/src/main/java/org/ballerinalang/net/http/FailoverClientHandshakeListener.java

```diff
@@ -0,0 +1,121 @@
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_PACKAGE_HTTP,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
+        WebSocketOpenConnectionInfo connectionInfo = new WebSocketOpenConnectionInfo(
+                wsService, webSocketConnection, webSocketClient);
+        webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
+        clientConnectorListener.setConnectionInfo(connectionInfo);
+        webSocketClient.set(WebSocketConstants.CLIENT_CONNECTOR_FIELD, webSocketConnector);
+        FailoverContext failoverConfig = (FailoverContext) webSocketClient.getNativeData(FAILOVER_CONFIG);
+        if (failoverConfig.isConnectionMade()) {
+            webSocketClient.set(WebSocketConstants.LISTENER_ID_FIELD, webSocketConnection.getChannelId());
+        } else {
+            WebSocketUtil.populateEndpoint(webSocketConnection, webSocketClient);
+        }
+        // Read the next frame when readyOnConnect is true in first connection or after the first connection
+        if (failoverConfig.isConnectionMade() || readyOnConnect) {
+            webSocketConnection.readNextFrame();
+        }
+        // Following these are created for future connection
+        // Check whether the config has retry config or not
+        // It has retry config, set these variable to default variable
+        FailoverContext failoverContext = ((FailoverContext) webSocketClient.getNativeData(FAILOVER_CONFIG));
+        if (hasRetryConfig(webSocketClient)) {
+            ((RetryContext) webSocketClient.getNativeData(RETRY_CONFIG)).setReconnectAttempts(0);
+            failoverContext.setFinishedFailover(false);
+        }
+        ArrayList targets = failoverConfig.getTargetUrls();
+        int currentIndex = failoverConfig.getCurrentIndex();
+        logger.info(CONNECTED_TO + targets.get(currentIndex).toString());
+        // Set failover context variable's value
+        failoverContext.setInitialIndex(currentIndex);
+        failoverContext.setConnectionMade();
+        countDownLatch.countDown();
+    }
+
+    @Override
+    public void onError(Throwable throwable, HttpCarbonResponse response) {
+        if (response != null) {
+            webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD, HttpUtil.createResponseStruct(response));
+        }
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_PACKAGE_HTTP,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
+        WebSocketOpenConnectionInfo connectionInfo = new WebSocketOpenConnectionInfo(
+                wsService, null, webSocketClient);
+        webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
+        countDownLatch.countDown();
+        if (throwable instanceof IOException || throwable instanceof BlockingOperationException) {
+            doAction(connectionInfo, throwable, null);
```

Comment:
Shall we improve the name of this method? `doAction()`? I feel like it is a very generic name.